### PR TITLE
Implement select command

### PR DESCRIPTION
**Pull request by harshavardhana**

### CODE REVIEW

**Comment by harshavardhana** [comment]
This PR is ready for review @kannappanr @nitisht

File: cmd/client-s3.go [hunk 1]
```diff
@@ -384,6 +386,61 @@ func (c *s3Client) ListNotificationConfigs(arn string) ([]notificationConfig, *p
...
+				FieldDelimiter:  ",",
+			},
+		}
+		opts.OutputSerialization = minio.SelectObjectOutputSerialization{
+			CSV: &minio.CSVOutputOptions{
+				RecordDelimiter: "\n",
+				FieldDelimiter:  ",",
+			},
+		}
+	} else if strings.Contains(origContentType, "json") {
```
Comment (kannappanr) [inline cmd/client-s3.go]: we can maybe skip if the file is not either one of `csv` and `json`

Comment (harshavardhana) [reply cmd/client-s3.go]: it is skipped not by this function instead by the caller...

Comment (kannappanr) [reply cmd/client-s3.go]: This is the output I see when I run the recursive command with one file which is neither csv not json
```
mc: <ERROR> Unable to run select Invalid data source type. Only CSV and JSON are supported at this time.
2.000000
1178008.000000
```
I think this tells the user that they have a non `csv and json` file. I guess this is good enough.

**Comment by harshavardhana** [comment]
There is a CLI question that needs review from @abperiasamy - blocked until then

**Comment by harshavardhana** [comment]
Unlocked `--expression` is changed to `--query`

**Comment by harshavardhana** [comment]
please review @kannappanr @sinhaashish

**Pull request merged**